### PR TITLE
New Docker and Travis Support

### PR DESCRIPTION
**Pull request by mbrandenburger**
This PR adds new FPC docker runtime and dev images with SGX SDK 2.6 support. Moreover, it adds support for Travis-CI builds.

### CODE REVIEW

File: common/enclave/generate_mrenclave.sh [hunk 1]
```diff
@@ -19,7 +19,6 @@ go_name=mrenclave.go
 cd $build_dir
 sgx_sign gendata -enclave enclave.so -config $enclave_dir/enclave.config.xml -out $tmp_name
 dd if=$tmp_name bs=1 skip=188 of=$hex_out_name count=32
-hexdump -C $hex_out_name
```
Comment (mbrandenburger) [inline common/enclave/generate_mrenclave.sh]: Removed as it only outputs the hexdata here but nothing more. Also hexdump was not available in new dockerbase image and decided not to add it.

File: Makefile [hunk 2]
```diff
@@ -30,7 +30,7 @@ gotools:
 	# install goimports if not present
 	$(GO) get golang.org/x/tools/cmd/goimports
 
-godeps:
+godeps: gotools
```
Comment (mbrandenburger) [inline Makefile]: goimports is needed to build `utils` but only added later.

File: common/enclave/generate_mrenclave.sh (hunk 1 above)
Comment (g2flyer) [reply common/enclave/generate_mrenclave.sh]: we might want at some point remove it completely and just rely on the -dumpfile option of sgx_sign (see issue #77)

File: .travis.yml [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+
+go_import_path: github.com/hyperledger-labs/fabric-private-chaincode
+
+env:
+  global:
+    # SGX config
+    - SGX_MODE=SIM
+    - SGX_SDK=/opt/intel/sgxsdk
+    # SGX SSL
+    - OPENSSL=1.1.0j
```
Comment (g2flyer) [inline .travis.yml]: We might upgrade to 1.1.0l which is the latest version?  (btw: apropos version, bruno noticed that you forgot to update README.md from 1.4.2 to 1.4.3, so if you change versions, you might fix also README.md. While version changes could deserve a separate pr, you already changed the sdk version, so we might also update openssl version and fix other version-related omissions ...)

File: Makefile [hunk 4]
```diff
@@ -6,7 +6,7 @@
 TOP = .
 include $(TOP)/build.mk
 
-SUB_DIRS = utils ercc ecc_enclave ecc tlcc_enclave tlcc examples integration # docs
+SUB_DIRS = docker utils ercc ecc_enclave ecc tlcc_enclave tlcc examples integration # docs
```
Comment (g2flyer) [inline Makefile]: I must admit i liked having the docker stuff in utils (given that we already have it and kept docker stuff previously there) rather than putting it in the top-level ....

File: docker/base/Dockerfile [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+    libssl-dev \
+    libcurl4-openssl-dev \
+    libtool \
+    cmake \
+    vim \
+ && apt-get -y -q upgrade \
+ && apt-get clean \
+ && rm -rf /var/lib/apt/lists/*
+
+ARG GO_VERSION=go1.13
```
Comment (g2flyer) [inline docker/base/Dockerfile]: For clarity, it would be nicer next to the GO_TAR definition.  Also, i guess right now we need go really only for dev? So why not just put it there?

On the other hand, right now we have sgxssl only in dev, once we would add deterministic build we would need it also in ``ccenv-build'' and so might need it in two different docker images?

File: docker/dev/Dockerfile [hunk 6]
```diff
@@ -0,0 +1,41 @@
+# Copyright IBM Corp. All Rights Reserved.
+#
+# SPDX-License-Identifier: Apache-2.0
+
+FROM fabric-private-chaincode/base:latest
+
+ARG OPENSSL=1.1.0j
```
Comment (g2flyer) [inline docker/dev/Dockerfile]: See comments in travis.yaml regarding version ...

File: docker/Makefile [hunk 7]
```diff
@@ -0,0 +1,24 @@
+# Copyright IBM Corp. All Rights Reserved.
+#
+# SPDX-License-Identifier: Apache-2.0
+
+TOP = ..
+include $(TOP)/build.mk
+
+FPC_DOCKER_NAMESPACE := fabric-private-chaincode
```
[the remark this answers is not in the document]
Comment (mbrandenburger) [reply docker/Makefile]: agree!

File: .travis.yml [hunk 8]
```diff
@@ -0,0 +1,69 @@
...
+
+addons:
+  apt:
+    packages:
+      - protobuf-compiler
+      - python-protobuf
+
+branches:
+  only:
+    - travis
```
[the remark this answers is not in the document]
Comment (mbrandenburger) [reply .travis.yml]: this restriction can be removed.

File: .travis.yml (hunk 3 above)
Comment (mbrandenburger) [reply .travis.yml]: Regarding fabric version, yes it seems that I didn't catch all 1.4.2 when updating README.md. Can one use variables in markdown?

File: Makefile (hunk 4 above)
Comment (mbrandenburger) [reply Makefile]: sure, we can keep it in utils

File: docker/base/Dockerfile (hunk 5 above)
Comment (mbrandenburger) [reply docker/base/Dockerfile]: ok, let's move go to dev image for the moment.

File: .travis.yml (hunk 3 above)
Comment (mbrandenburger) [reply .travis.yml]: Upgraded openssl and added this to commit msg

Comment (mbrandenburger) [reply .travis.yml]: with openssl 1.1.0j CI-build goes through
